public/app/features/explore/utils/links.ts

```diff
@@ -127,10 +128,34 @@ export const getFieldLinksForExplore = (options: {
         }
         return linkModel;
       } else {
+        let internalLinkSpecificVars: ScopedVars = {};
+        if (link.internal?.transformations) {
+          const fieldValue = field.values.get(rowIndex);
+          link.internal?.transformations.forEach((transformation) => {
+            if (transformation.type === 'regex' && transformation.expression) {
+              const regexp = new RegExp(transformation.expression);
+              const matches = fieldValue.match(regexp);
+              if (matches?.length > 0) {
+                internalLinkSpecificVars[transformation.variable || field.name] = {
+                  value: matches[1] || matches[0], // not global - return capture group if found, else full match
+                };
+              }
+            } else if (transformation.type === 'logfmt') {
+              const logFmtVal = logfmt.parse(fieldValue) as { [key: string]: string };
```

Comment:
@ifrost Do we need to do a type assertion when the library doesn't is ambiguous about what it returns ([src](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/logfmt/index.d.ts#L42)), or is there another way?

Reply:
I had a quick look at https://github.com/csquared/node-logfmt/blob/master/lib/logfmt_parser.js and it seems like in some cases in may return non-string values (`true`, `false`, `null`). Maybe we can use the provided type and treat values as `object` but stringify them? (we have `safeStringifyValue` function somewhere in utils, maybe we could use that)

Reply:
@ifrost that makes sense. I used the same pattern as [here](https://github.com/grafana/grafana/blob/main/public/app/features/variables/utils.ts#L65) though, where we only use this function if its a non-string - otherwise it adds quotes around a string value that we probably don't want.

One thing I noticed with `safeStringifyValue` though is that it would return an empty string with `false` instead of returning `"false"` like we would want. I changed the logic in the function - I can't imagine anyone wanting it to convert `true` but return an empty string on `false`. Seems way more likely to be a bug. Let me know what you think.